xonsh/prompt/base.py

```diff
@@ -24,8 +24,70 @@
 from xonsh.prompt.gitstatus import gitstatus_prompt
 
 
+@xt.lazyobject
+def DEFAULT_PROMPT():
+    return default_prompt()
+
+
+class PromptFormatter:
+    """Class used by base_shell"""
+    def __init__(self):
+        self.cache = {}
+
+    def __call__(self, template=DEFAULT_PROMPT, fields=None):
+        """Formats a xonsh prompt template string."""
+        if fields is None:
+            self.fields = builtins.__xonsh_env__.get('PROMPT_FIELDS', PROMPT_FIELDS)
+        else:
+            self.fields = fields
+        try:
+            prompt = self._format_prompt(template=template)
```

Comment:
I think only this line should be in the try-block.  Please move the other two lines to after the except.